python/cm/upgrade.py

```diff
@@ -256,6 +256,24 @@ def rpm_cmp_reverse(obj1, obj2):
         return res
 
 
+def recount_issues(obj, upgrade):
+    """
+    Recount of issues from the bottom up for all related objects component-service-cluster
+    or host-provider, or for cluster or provider objects
+    """
+    if obj.prototype.type == 'cluster':
+        for proto in Prototype.objects.filter(bundle=upgrade.bundle, type='service'):
```

Comment:
This loop is unnecessary. You can get services straight from ClusterObject

Reply:
Fixed